### PR TITLE
only show report when assembly is done

### PR DESCRIPTION
**Pull request by yunfangjuan**
# Description

1. only show report when assembly is done for pipeline 3+ 
2. download the taxid fasta based on assembly results
3. change nonhost fasta download to use (refined_)taxid_annot.fa

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix a feature that knowingly causes existing functionality to not work as expected)

# Test and Reproduce
Outline the steps to test or reproduce the PR here.

1. 

## Impacted Areas in Application
List general components of the application that this PR will affect:

Which pages?

- [ ] Login Screen
- [ ] Single Upload Page
- [ ] Batch Upload Page
- [ ] All Projects Page
- [ ] Sample Details Page
- [ ] Report Page

Specific components 

* 

# Testing Script:

###Login Page
* Successfully login to the application and get taken to the project page

###Single Upload
* Using an AWS path that doesn't exist throws an error
* Succesfully uploaded files from a folder (use test host)
* Uploading a duplicate file throws an error
* Pipeline gets initiated successfully 

###Batch Upload
* Using an AWS path that doesn't exist throws an error
* Succesfully uploaded files from a folder (use test host)
* Navigate to batch upload settings page after upload 
* Host selection from previous page still selected 
* Uploading a duplicate file throws a warning
* Pipeline gets initiated successfully 

###All Project Page
* Switch between projects 
* Search for a sample
* Navigate through different pages of samples
* Select Upload new sample button
* Select sample to see generated report

###Sample Details Page
* Information is consistent from Project Page
* Notes are editable and save successfully 
* Can navigate to report page 

###Report Page
* Data loads succesfully 
* Collapse and Uncollapse button works
* Clicking on the taxonomy name gives you total reads for that taxon
* Category filter works
* Genus Search works
* Filters work as expected
* Disabling Filters works
* Filtering by column works 

# Checklist:

- [ ] I have run through the testing script to make sure current functionality is unchanged
- [ ] I have done relevant tests that prove my fix is effective or that my feature works
- [ ] I have spent time testing out edge cases for my feature
- [ ] I have updated the test script or pull request template if necessary
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by cdebourcy** [comment]
Is there a way we could make this more future-proof? From past experience, dealing with these name prefixes is really error-prone. How about we make a clean break and leave the file names the same while the meaning changes? Or how about we store these output paths in the pipeline_runs table when we create a pipeline_run (based on pipeline_version), and never change them again? So we can avoid dealing with this complexity in the code and just always call the same column in the table?

File: app/models/pipeline_run.rb [hunk 1]
```diff
@@ -400,7 +400,7 @@ def db_load_taxon_counts
   end
 
   def db_load_refined_taxon_counts
-    output_json_s3_path = "#{postprocess_output_s3_path}/#{REFINED_TAXON_COUNTS_JSON_NAME}"
+    output_json_s3_path = s3_file_for("refined_taxon_counts")
```
Comment (MarkAZhang) [inline app/models/pipeline_run.rb]: refined_taxon_counts doesn't seem to exist in `s3_file_for` any more. Did you mean `taxon_counts`?

File: app/lib/report_helper.rb [hunk 2]
```diff
@@ -238,15 +238,13 @@ def all_categories
     ALL_CATEGORIES
   end
 
-  def fetch_taxon_counts(pipeline_run_id, background_id, refined = false)
+  def fetch_taxon_counts(pipeline_run_id, background_id, _refined = false)
```
Comment (MarkAZhang) [inline app/lib/report_helper.rb]: Is `_refined` used anywhere?

File: app/models/pipeline_run.rb (hunk 1 above)
Comment (MarkAZhang) [reply app/models/pipeline_run.rb]: Ah I see. I guess this loader doesn't get called any more.

Can you add some comments with `TODOs` just so others don't get confused?

**Pull request merged**